### DISCUSSION TITLE
Unguided Campaign - Improving C004L007.DAT

### DISCUSSION THREAD

**Discussion by Moburma** [swfans/syndwarsfx#198] [Show and tell]
So, more low hanging fruit. This is quite a good level, but a barely finished one. It's another pre-alpha level, and quite revealing due to a few things. Firstly, it's before the PING invisible commands were created, so the vehicle you're pursuing can't actually leave the map, just awkwardly hang around there if it gets away. Secondly the zealots all have pointless Disrupters; I think based on other levels there was going to be some kind of baseline template where they would regularly be equipped with them to to foil mass-persuading strategies, but the AI was never written to actually use the disrupter intelligently so they're just useless.

Anyway, I spruced up what was there, but feel like it needs a little more. The big problem before was that the bad guy would take off in his APC when you got close, enticing you to get into an exciting chase with him to his flying getaway vehicle. Only umm, you couldn't, as there were no vehicles in the level to board, meaning if you didn't kill him ASAP you'd lose. It also rewarded cheesy strategies like rushing for HIS car and getting in it. While I don't think this should be prevented, it felt lame and anti-fun. To that end, I've added a new vehicle at the start for the player to use. 

This actually starts off occupied, to stop cheesy stuff like just going straight to the landing pad and stealing the baddies' flying vehicle, making the level a doddle. Once the enemies start to move, the occupier will leave the bike, meaning the player can sprint over and use it to chase them before they get away. 

This part I'm mostly happy with. The other problem is what next? It's a pretty short level, and it's likely the idea was the player would next go and kill the zealots, as aside from them plain being there, there is a really cheesy ambush from some police waiting for the player on the way. This all feels a bit crude though, feels like it should tightened up and made more interesting.

Summary:

- Updated to modern level format, and Ping commands used for hiding the getaway car if the player fails
- Disruptors removed from zealots
- new bike added at the start the player can use if they don't kill the soldier before he boards his vehicle
- Should there be a bit of a pause before the enemy boards his flight to give the player a chance? Maybe it's too hard to use the car to get there
- Zealot part feels rough. Need to turn off the WITHIN AREA command on them if the player gets close as the fight is too easy and looks weird
- Police ambush looks a bit obvious and crap, maybe hide some of them? 
- Maybe need to beef up characters - this seems to have been before mods existed/were in final game format
- What else should go in the level? Is it already enough or does it need more action?

Miss002.ini update:

[missuccess1]
P00 = GAME_OBJ_ALL_G_DEAD( Group(13) )
P01 = GAME_OBJ_ALL_G_DEAD( Group(6) )

File download:

https://easyupload.io/pnnlp1

**Comment by mefistotelis** [comment]
What if one of the police had an Ion Mine? You can't pick it up - your punks don't have any. But you can kill him at the proper spot so he drops the mine on the road.

Another thing which seem obvious - getting to the bank from the back. This should be somehow rewarded, ie. if you get there, you can blow the bank and kill the militia without direct combat. But you' have to receive high explosive, or be able to self-destruct one of yours. Another idea would be the ability to blow up the bezier road.
EDIT:
Thinking about it more, the reward from getting to the bank from back doesn't have to solve the situation, it could be just a scene - ie. an Unguided spawns at edge of the map, runs towards the bank, and attacks the militia with an Uzi. Then gets disintegrated with one shot.

White women seem to like your team very much; they go towards your team, not running into them only because getting scared of the weapons. They don't seem to have commands for that, so I assume they're in the same group as either police or militia?

**Reply by Moburma** [reply]
Hmm, not sure about the combat stuff, I think the start of the mission is ok as-is, I just felt that the soldier getting away was too drastic as there was no way to win then. I think adding a car solves that. Also it means the player can then upgrade to the abandoned flying vehicle (although that has its own issues with balance).

I know what you mean about the white women (another revealing thing about this mission is it's clear the other civilian varieties didn't exist yet) thing. The group relations on this level are weird and just seem to be left over from a totally different mission (so are the group names). The civilians are guardians of other groups which makes no sense, so I'll clean all that up. There are also weird (I think illegal) settings of Guardians for group 255, which just seem like random junk, don't think that is legit in the game (in my editor I clamped from 0-31, but this isn't enforced if you don't browse the groups, so is still as it was in the original level file currently).

EDIT: Speaking of Guardians/groups I really need to sit down and properly figure out how to use these attributes. It's clear looking at the levels that Bullfrog themselves did not know how they were meant to be used! There are all kinds of weird and contradictory usage of them, like setting them on the player group (in fact there is all kinds of stuff set against the player group in levels that is nonsense - the player decides how his group behaves). It's not possible to understand how the guardian field was meant to be used just from the level files as there is so much obviously wrong stuff set. I am pretty sure it's meant to be Guardian set = THIS group acts as a guardian for THAT group, i.e. will act as if they had been shot on themselves if the other group is attacked. It's not apparent if this NEEDS Truce set for that group (makes sense it does) or if it implicitly includes that setting by them being a guardian.

**Reply by Moburma** [reply]
I fixed the white women problem (insert your own joke here). It was quite weird and underlines that again I need to sit down and model all the different outcomes to actually understand how guardian is meant to be used. In this case it was fortuitous that I'd removed all the settings for group 3 (the women's group) already, but it was still happening. What the actual issue was turned out to be that because group 8 (the main soldier group) was guardian of group 3 (women group). My understanding was that it would mean group 8 would defend group 3 if group 3 was shot at, but actually it seems to be the _reverse_, so that group 3 was defending group 8 when group 8 was shot at. So you were right, you were getting thronged by all these women as they were trying to fight you for shooting the soldiers. Weird.

Another dopey thing I see set a lot is that a group is set as truced to itself. Makes zero sense (and clearly isn't needed), unless this is a pre-alpha throwback to when it was needed.

**Comment by geist22** [comment]
My distaste for NPCs getting inside vehicles, driving off the map and failing your mission is something I've already mentioned in the past. Unless you have the means of effectively cutting off possible escape routes beforehand, such missions are practically guaranteed to fail and rely on you using your knowledge as the player to prevent such events from happening in future attempts, which in-universe can only be described as the Executive knowing the future.

My only successful way to clear this mission was making a run for one of the enemies' vehicles and driving to the flying vehicle straight away, unless they stopped escaping in favor of attacking my squad. The bike you added doesn't work in practice, for two reasons: first, with it being occupied, the player has no way of knowing that it would coincidentally become available the moment you trigger the enemies to run. It doesn't really make much sense either, and feels too artificial. Second, even if you know, it doesn't work out. I did one attempt where I got close enough to the enemies to trigger their escape, then instantly ran back to the vehicle and drove to their destination. By the time my bike cleared the gate, the target was just about to enter the flying vehicle. The way back to the bike is too long, and you also get slowed down by other, civilian vehicles on your way. 

Speaking of vehicles, that traffic jam that seems to be caused by the APCs being parked on the road rather than next to it, I guess this is just a leftover from what you've managed to salvage from the original and will be cleared up later? Also, there are multiple police vehicles and at least one big green APC all with civilians inside them. Those should be replaced with other vehicles, as civilians don't normally use them. There's also a church vehicle with a civilian, but we can see this in the campaign every now and then, so it's nothing out of the ordinary. The flying escape vehicle is a church vehicle too, it might make sense to replace it with a green APC which is more in line with the theme of the syndicate.

I like the suggestion about Ion Mines. I feel that consumbes in general are a bit underused, at least I rarely use them, because doing so when you don't have to feels like a waste of money. With Unguided being weak and slow, and money/pickups not carrying over between missions, the Unguided campain could be an opportunity to make more use of consumables.
But killing police to drop Ion Mines in a specific spot seems too finicky. Perhaps the walled off area in the east could be used as a warehouse occupied by the syndicate, with a few mines inside for you to pick up. And either a vehicle or a High Explosive to enter. But as mentioned above, with a High Explosive you could just blow up the bank instead. A vehicle could be made available from the start, and the area with the flying vehicle could be guarded by multiple agents to prevent or discourage you from getting there, although that would also prevent you from using it for the later part of the mission, in case that is intended as an available option.

**Reply by Moburma** [reply]
Good points, one thing I would say is do we think overall it is even a level worth saving? I like the basic idea of a chase against some kind of general (rather than just more zealots/agents), but I take your point it is heavily reliant on trial and error. It's a bit of a weird level, and I do wonder if the idea is too thin/boxes it self in too much. It's quite simple if you know what to do (kill majority of soldiers from a distance, then sprint to the big APC before the target can = easy win), and the chase part is hard to make fair (you're at the mercy of random traffic, and vehicle to vehicle combat doesn't really work in this game). The level also has nowhere to go outside that, the zealot stuff feels really tacked on. Basically would it be better to dump it for an original level? Particularly this map is crying out for vehicle/flying vehicle shenanigans that this level barely touches on

>Speaking of vehicles, that traffic jam that seems to be caused by the APCs being parked on the road rather than next to it, I guess this is just a leftover from what you've managed to salvage from the original and will be cleared up later? 

Yeah it is, it's worth playing the original level to see the (few) differences. Again I'm not even sure this _can_ be fixed, I think I can just move the cars out of the way into parked positions, but not 100% sure that's possible on this type of road. I saw the weird stuff like police cars not actually having police in, that can easily be fixed later but is indicative of how crude this level is in many ways

>Ion Mines.

I think you've both warmed me up to that, so I'll try and incorporate them. Probably makes sense to start with them.

>New Bike

I think that needs changing too, fair point it's not clear to the player they can use it. It would make more sense to just place an empty vehicle literally next to the bank so that a) it can be seen from minute one it's vacant, and b) it triggers the enemies if the player goes to get it, so you can't just rush straight to the end, solving that problem.

Anyway, I think the bigger issue is more is the level actually salvageable? There's not that much to it. I want to try and include as many Bullfrog created levels as possible, but it's clear a) most of them don't really have much rhyme or reason to what's going on and b) they are most technically really poor/simple as they were made so early on and before so many commands and techniques were possible (it's obvious that most of the final game levels went through a LOT of revisions, looking at the level files) . As such I do wonder if it would be better to be more ruthless on a lot of the pre-alpha levels, this one has always been a bit borderline whether it's good or not in my mind.

**Reply by geist22** [reply]
> the zealot stuff feels really tacked on. Basically would it be better to dump it for an original level? Particularly this map is crying out for vehicle/flying vehicle shenanigans that this level barely touches on

One idea would be to remove the zealots from the objectives so they're not required to kill anymore and the mission ends with the militia dead, and instead put either a few Ion Mines inside their temples or an empty flying church vehicle somewhere between them, so the player can either make the run for the APCs or assault the temple and obtain a slower but safer method.

**Comment by mefistotelis** [comment]
> The way back to the bike is too long, and you also get slowed down by other, civilian vehicles on your way.

Yes, I agree to that. And if I didn't knew that it's getting freed, I'd never figure that out.

A more dynamic scene is needed: the car wanders on the map, or just travels between two points to make sure it's close enough. When you're close to the bank, the car parks and the driver exits, somewhere where the player has a chance to see it.

That would be a lot better than exiting a car off screen, far away from the event which triggers it.

EDIT:
For solving the traffic jam:
* cars without drivers should not block other cars
* just being parked should also be enough for the cars to pass

Not sure if the car is parked too close to the road nodes, or it has some flags which make it blocking. Most likely both - cars without drivers normally have collisions disabled no matter where they're located, but the parking place should also be distant enough from the traffic node that it should allow the cars to pass.

**Reply by geist22** [reply]
There are instances where parked cars are used as blockades - for example Church mission 10: Bahrain. Not that it matters in this particular case, as there are no regular cars available.

**Comment by Moburma** [comment]
Ok, new version, taking onboard feedback above:

- Large APC moved onto kerb - this now allows the traffic to flow normally, but arguably it makes the level very slightly harder as the target now has less distance to travel to get in his vehicle and escape. I think this may prevent the "steal his car" strategy to win
- 1st Unguided now has ion mines. Maybe I will give them to everyone to encourage playing about, but it probably makes it a bit too easy that way (i.e. mine every inch of road in or out)
- Bike has now become a spare APC next to the bank. This is unpopulated and can be used by the player at any time
- Police cars now actually have armed police in them. Need to think what to do with the other vehicles, do we want armed soldiers in the other military APCs?

Not fully tested it yet, so see what you think.

Download:
https://easyupload.io/77hr4t

**Reply by geist22** [reply]
I've tried to get one of the militia to shoot a policeman on accident to incite a fight between the factions, but it seems they are incapable of hitting each other. Are they on the same team? This doesn't make sense to me.

I've just barely succeeded at killing the targets before they board the flying vehicle after a car chase with the new APC, though I don't know how reliably this works as I've only done it once.
For the Ion Mines, I felt that there's just too much traffic to use them strategically and had to run in front of the fleeing APCs and drop them right there, which also got some of my Unguided killed. The traffic could be cut down about half of what it is currently.

The zealots still have this weird behaviour where they move back and forth just outside your range instead of engaging. Though I remember seeing the same behaviour in one of the original campaign maps, maybe even in the same spot.

**Reply by Moburma** [reply]
> I've tried to get one of the militia to shoot a policeman on accident to incite a fight between the factions, but it seems they are incapable of hitting each other. Are they on the same team? This doesn't make sense to me.

Yes, they're truced. It is a bit weird, I get the impression a lot of the group relations on this level are a template/leftover from a different level. Not sure if it would be a good idea or not to reward getting them fighting the soldiers as there are quite a few police, but I will try it. Actually one thing you CAN do is use the crap police ambush to get them to fight the zealots, as they are set to shoot if gun out on them. The only problem is by then you've usually killed most of the police and they don't last long against electron maces.

> I've just barely succeeded at killing the targets before they board the flying vehicle after a car chase with the new APC, though I don't know how reliably this works as I've only done it once. For the Ion Mines, I felt that there's just too much traffic to use them strategically and had to run in front of the fleeing APCs and drop them right there, which also got some of my Unguided killed. The traffic could be cut down about half of what it is currently.

Yeah, I think the target needs to pause when he gets to the getaway point before leaving his car so the player has time to catch up. I also wonder if the objective should be just to kill HIM and not the rest. I also wonder if this level really works at all, is it actually fun!

I had no problems with the ion mines, I just didn't use them until only the last 3 soldiers are left. The problem there is anticipating this, and arguments for against trial and error etc. Could thin out the traffic though, it's a bit over the top regardless.

> The zealots still have this weird behaviour where they move back and forth just outside your range instead of engaging. Though I remember seeing the same behaviour in one of the original campaign maps, maybe even in the same spot.

This is the result of using a WITHIN_AREA command. I flagged this earlier, basically using this command puts up an invisible wall around the NPC they can never leave. This works great for patrols and in this instance for making them walk around the central pillar like they are doing some weird religious ritual. The trouble is it is awful for combat, as they can't leave it to fight, and their AI doesn't know how to deal with the restriction, so they try to flee to cover they can't get to, etc. What I need to do is give them a clause like WAND_MEM_G_NEAR so that they stop when the player gets too close and then turn off WITHIN_AREA so they can fight properly. Haven't done it yet as I was pondering what to do with the zealot part entirely.
